.github/actions/create-project-issue-v1/src/run.test.ts

```diff
@@ -0,0 +1,389 @@
+      getInput.withArgs('body', { required: true }).throws({
+        message: 'Input required and not supplied: body'
+      })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(
+        setFailed.calledWith('Input required and not supplied: body')
+      )
+    })
+  })
+
+  describe('when the `project_number` input is not a number', () => {
+    it('throws an error', async () => {
+      generateInputs({ project_number: 'foo' })
+
+      const core = {
+        getInput,
+        setFailed
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.calledOnce)
+      assert.isTrue(setFailed.calledWith('project_number must be a number'))
+    })
+  })
+
+  describe('when the `repository` input is not provided', () => {
+    it('uses the `github.context.repo` to create an issue and move it to board', async () => {
+      generateInputs({ repository: '' })
+
+      await createAndMoveIssueStub()
+
+      const core = {
+        getInput,
+        setFailed,
+        setOutput,
+        info
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.notCalled)
+      assert.equal(info.callCount, 6)
+      assert.isTrue(setOutput.calledOnce)
+      assert.isTrue(
+        setOutput.calledWith('issue_url', 'https://deque.bizzy.com')
+      )
+    })
+  })
+
+  describe('when the `repository` input is provided', () => {
+    it('uses the `repository` to create an issue and move it to board', async () => {
+      generateInputs({ repository: 'repo' })
+
+      await createAndMoveIssueStub()
+
+      const core = {
+        getInput,
+        setFailed,
+        setOutput,
+        info
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.notCalled)
+      assert.equal(info.callCount, 6)
+      assert.isTrue(setOutput.calledOnce)
+      assert.isTrue(
+        setOutput.calledWith('issue_url', 'https://deque.bizzy.com')
+      )
+    })
+  })
+
+  describe('when the `labels` input is provided', () => {
+    it('creates an issue with the labels', async () => {
+      generateInputs({ labels: 'foo,bar' })
+
+      await createAndMoveIssueStub()
+
+      const core = {
+        getInput,
+        setFailed,
+        setOutput,
+        info
+      }
+
+      await run(core as unknown as Core, github as unknown as GitHub)
+
+      assert.isTrue(setFailed.notCalled)
+      assert.equal(info.callCount, 6)
+      assert.isTrue(setOutput.calledOnce)
+      assert.isTrue(
+        setOutput.calledWith('issue_url', 'https://deque.bizzy.com')
+      )
+    })
+  })
+
+  describe('when the `assignees` input is provided', () => {
+    it('creates an issue with the assignees', async () => {
+      generateInputs({ assignees: 'foo,bar' })
```

Comment:
This test doesn't verify the integration that the issue created used the assignees `foo` and `bar`. You'll need to look at the `getOctokit` create issue stub to verify it used the params.

I would like to see integration tests that verify the inputs provided to the `run` action are being used all the way through the code and not just that you got an issue url back. These tests include:

* verifying that the inputs `title`, `body`, `assignees`, `labels`, and `repository` were all used when creating the issue
* verifying that the inputs `column_name` and `project_number` were used when adding the issue to the project board. the mocks sort of verify this but it is not a direct `assert`, meaning that when it fails the error message will not make sense to fix the problem (the thing that will fail will be something like `info.callCount not called 6 times`. This isn't a helpful message to resolve the problem of the mock not being used because the inputs didn't match)